docs/useCreate.md

```diff
@@ -61,6 +61,267 @@ const LikeButton = () => {
+
+**Tip**: In react-admin components that use `useCreate`, you can override the mutation options using the `mutationOptions` prop. This is very common when using mutation hooks like `useCreate`, e.g., to display a notification or redirect to another page.
+
+For instance, here is a button using `<Create mutationOptions>` to notify the user of success using the bottom notification banner:
+
+{% raw %}
+```tsx
+import * as React from 'react';
+import { useNotify, useRedirect, Create, SimpleForm } from 'react-admin';
+
+const PostCreate = () => {
+    const notify = useNotify();
+    const redirect = useRedirect();
+
+    const onSuccess = (data) => {
+        notify(`Changes saved`);
+        redirect(`/posts/${data.id}`);
+    };
+
+    return (
+        <Create mutationOptions={{ onSuccess }}>
+            <SimpleForm>
+                ...
+            </SimpleForm>
+        </Create>
+    );
+}
+```
+{% endraw %}
+
+## Return Value
+
+The `useCreate` hook returns an array with two values:
+
+- the `create` callback, and
+- a mutation state object with the following properties:
+  - `data`,
+  - `error`,
+  - `isError`,
+  - `isIdle`,
+  - `isPending`,
+  - `isPaused`,
+  - `isSuccess`,
+  - `failureCount`,
+  - `failureReason`,
+  - `mutate`,
+  - `mutateAsync`,
+  - `reset`,
+  - `status`,
+  - `submittedAt`,
+  - `variables`.
+
+The `create` callback can be called with a `resource` and a `param` argument, or, if these arguments were defined when calling `useCreate`, with no argument at all:
+
+```jsx
+// Option 1: define the resource and params when calling the callback
+const [create, { isPending }] = useCreate();
+const handleClick = () => {
+    create(resource, params, options);
+};
+
+// Option 2: define the resource and params when calling the hook
+const [create, { isPending }] = useCreate(resource, params, options);
+const handleClick = () => {
+    create();
+};
+```
+
+For a detailed description of the mutation state, check React-query's [`useMutation` documentation](https://tanstack.com/query/v5/docs/react/reference/useMutation).
+
+Since `useCreate` is mainly used in event handlers, success and error side effects are usually handled in the `onSuccess` and `onError` callbacks. In most cases, the mutation state is just used to disable the save button while the mutation is pending.
+
+## `mutationMode`
+
+The `mutationMode` option lets you switch between three rendering modes, which change how the success side effects are triggered:
+
+- `pessimistic` (the default)
+- `optimistic`, and
+- `undoable`
+
+**Note**: For `optimistic` and `undoable` modes, the record `id` must be generated client side. Those two modes are useful when building local first applications.
+
+Here is an example of using the `optimistic` mode:
+
+```jsx
+// In optimistic mode, ids must be generated client side
+const id = uuid.v4();
```

Comment:
I was wondering if it was common to do it like this, generating the id client side. I understand the why, but I'm not sure on the how we solve this. Couldn't we have a mechanism witch a temporary id, replacing it as the data is fetched ?

Reply:
Yes it is common in offline/local first scenarios. A temporary mechanism would make handling relationships such as those with ReferenceOneInput or ReferenceManyToManyInput a nightmare to handle. This is a good starting point and if we do find a way to use temporary id, we can add it later.